src/flashbar/collapsible-flashbar.tsx

```diff
@@ -188,106 +187,164 @@ export default function CollapsibleFlashbar({ items, ...restProps }: FlashbarPro
   const getItemId = (item: StackableItem | FlashbarProps.MessageDefinition) =>
     item.id ?? (item as StackableItem).expandedIndex ?? 0;
 
-  // This check allows us to use the standard "enter" Transition only when the notification was not existing before.
-  // If instead it was moved to the top of the stack but was already present in the array
-  // (e.g, after dismissing another notification),
-  // we need to use different, more custom and more controlled animations.
-  const hasEntered = (item: StackableItem | FlashbarProps.MessageDefinition) =>
-    enteringItems.some(_item => _item.id && _item.id === item.id);
-  const hasLeft = (item: StackableItem | FlashbarProps.MessageDefinition) => !('expandedIndex' in item);
-  const hasEnteredOrLeft = (item: StackableItem | FlashbarProps.MessageDefinition) => hasEntered(item) || hasLeft(item);
-
-  const showInnerContent = (item: StackableItem | FlashbarProps.MessageDefinition) =>
-    isFlashbarStackExpanded || hasLeft(item) || ('expandedIndex' in item && item.expandedIndex === 0);
-
-  const shouldUseStandardAnimation = (item: StackableItem, index: number) => index === 0 && hasEnteredOrLeft(item);
-
-  const getAnimationElementId = (item: StackableItem) => `flash-${getItemId(item)}`;
-
-  const renderList = () => (
-    <ul
-      ref={listElementRef}
-      className={clsx(
-        styles['flash-list'],
-        isFlashbarStackExpanded ? styles.expanded : styles.collapsed,
-        transitioning && styles['animation-running'],
-        initialAnimationState && styles['animation-ready'],
-        isVisualRefresh && styles['visual-refresh']
-      )}
-      id={flashbarElementId}
-      aria-label={ariaLabel}
-      aria-describedby={isCollapsible ? itemCountElementId : undefined}
-      style={
-        !isFlashbarStackExpanded || transitioning
-          ? {
-              [customCssProps.flashbarStackDepth]: stackDepth,
-            }
-          : undefined
-      }
-    >
-      <TransitionGroup component={null}>
+  const renderListWithoutAnimations = () => {
```

Comment:
This is too much code copy paste isn't it? It can get our of sync very easily

Any options how reduce this duplication?

Reply:
Grouped the common props together

Reply:
I added a test which compares the HTML output. Note:
- For this I needed the test to remove some a11y-related attributes that are not relevant to the visual result, because they have dynamic values
- Only for the collapsed state. For the expanded state, it gets trickier since we do need to add classes for the motion case, for the animations to work

If we want to be even more precise here, my suggestion would be to consider actually making screenshot tests with motion enabled too for this component.